File: project/app/src/main/java/org/owntracks/android/injection/modules/ActivityModules/RegionActivityModule.java
package org.owntracks.android.injection.modules.ActivityModules;

import android.support.v7.app.AppCompatActivity;

import org.owntracks.android.injection.modules.BaseActivityModule;
import org.owntracks.android.injection.scopes.PerActivity;
import org.owntracks.android.ui.region.RegionActivity;
import org.owntracks.android.ui.region.RegionMvvm;
import org.owntracks.android.ui.region.RegionViewModel;

import dagger.Binds;
import dagger.Module;

@Module(includes = BaseActivityModule.class)
public abstract class RegionActivityModule {

    @Binds
    @PerActivity
    abstract AppCompatActivity bindActivity(RegionActivity a);

    @Binds abstract RegionMvvm.ViewModel bindViewModel(RegionViewModel viewModel);
}

File: project/app/src/main/java/org/owntracks/android/injection/modules/android/ActivityModules/RegionActivityModule.kt
package org.owntracks.android.injection.modules.android.ActivityModules

import androidx.appcompat.app.AppCompatActivity
import dagger.Binds
import dagger.Module
import org.owntracks.android.injection.scopes.PerActivity
import org.owntracks.android.ui.region.RegionActivity
import org.owntracks.android.ui.region.RegionMvvm
import org.owntracks.android.ui.region.RegionViewModel

@Module(includes = [BaseActivityModule::class])
abstract class RegionActivityModule {
    @Binds
    @PerActivity
    abstract fun bindActivity(a: RegionActivity?): AppCompatActivity?

    @Binds
    abstract fun bindViewModel(viewModel: RegionViewModel): RegionMvvm.ViewModel<RegionMvvm.View>
}